### PR TITLE
feat(dropdownmenu): css changes and data attrs

### PR DESCRIPTION
**Pull request by Barsnes**
part of #2295

### CODE REVIEW

File: packages/react/src/components/DropdownMenu/DropdownMenuGroup.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const DropdownMenuGroup = forwardRef<
         {...(heading ? { 'aria-labelledby': headingId } : {})}
         ref={ref}
         role='group'
-        className={'ds-dropdownmenu__section'}
+        className={'ds-dropdownmenu__group'}
```
Comment (eirikbacker) [inline packages/react/src/components/DropdownMenu/DropdownMenuGroup.tsx:28]: should expose `className={cl('ds-dropdownmenu', className)}` here too? ☺️

Comment (Barsnes) [reply packages/react/src/components/DropdownMenu/DropdownMenuGroup.tsx:28]: We try to put their `className` and `style` on the outermost element, which in this case is the `li` element

Comment (eirikbacker) [reply packages/react/src/components/DropdownMenu/DropdownMenuGroup.tsx:28]: oh, sorry - did not see this! carry on :D

File: packages/css/dropdownmenu.css [hunk 2]
```diff
@@ -1,10 +1,6 @@
 .ds-dropdownmenu {
-  --dsc-dropdownmenu-background: var(--ds-color-neutral-background-default);
-  --dsc-dropdownmenu-padding: var(--ds-spacing-2);
-  --dsc-dropdownmenu-border-radius: min(1rem, var(--ds-border-radius-md));
-  --dsc-dropdownmenu-border: 1px solid var(--ds-color-neutral-border-subtle);
+  --dsc-dropdownmenu-padding: var(--ds-spacing-3) var(--ds-spacing-2);
   --dsc-dropdownmenu-min-width: 260px;
-  --dsc-dropdownmenu-shadow: var(--ds-shadow-md);
   --dsc-dropdownmenu-hr-border-color: var(--ds-color-neutral-border-subtle);
```
Comment (eirikbacker) [inline packages/css/dropdownmenu.css:4]: Maybe we should have a convention here - `--dsc-dropdown-divider` like in `--dsc-table-header-divider` or something? Also, there are no stories showing usage of hr in Dropdown? Maybe we even can remove `--dsc-dropdownmenu-hr-border-color` entirely now that `Divider` is subtle by default? ☺️

File: packages/css/dropdownmenu.css [hunk 3]
```diff
@@ -14,30 +10,25 @@
...
-  --dsc-dropdownmenu-min-width: 240px;
-}
 
-.ds-dropdownmenu--md {
-  --dsc-dropdownmenu-padding: var(--ds-spacing-3) var(--ds-spacing-2);
-  --dsc-dropdownmenu-min-width: 260px;
-}
+  &[data-size='sm'] {
+    --dsc-dropdownmenu-padding: var(--ds-spacing-2);
+    --dsc-dropdownmenu-min-width: 240px;
```
Comment (eirikbacker) [inline packages/css/dropdownmenu.css:21]: Can this be `rem` so we meet accessibility requirements for text zoom? (also on `data-size="lg"`)

Comment (Barsnes) [reply packages/css/dropdownmenu.css:21]: @Thunear Is `15rem` fine here?

File: packages/css/dropdownmenu.css (hunk 2 above)
Comment (Barsnes) [reply packages/css/dropdownmenu.css:4]: up to @Thunear to decide here, I don't think we need this anymore either

Comment (Thunear) [reply packages/css/dropdownmenu.css:4]: I agree. We should use the `Divider` component inside these types of components.

File: packages/css/dropdownmenu.css (hunk 3 above)
Comment (Thunear) [reply packages/css/dropdownmenu.css:21]: Yes, that should be fine.

Comment (Barsnes) [reply packages/css/dropdownmenu.css:21]: I did some rounding, and we now have

| size| old| new|
|--------|--------|--------|
| sm| 240px | 15rem |
| md | 260px | 16rem |
| lg | 280px | 18rem |

**Review by eirikbacker — APPROVED** [review]

**Pull request merged**